packages/fx-core/src/question/other.ts

```diff
@@ -942,6 +957,21 @@ export function oauthTokenUrlQuestion(): TextInputQuestion {
     title: getLocalizedString("core.addAuthActionQuestion.OAuthTokenUrl.title"),
     type: "text",
     cliDescription: "Token Url for oauth.",
+    validation: {
+      validFunc: (input: string): string | undefined => {
+        if (input.trim() === "") {
+          return getLocalizedString("core.addAuthAction.validation.url");
+        }
+
+        try {
+          new URL(input);
+        } catch (error) {
+          return getLocalizedString("core.addAuthAction.validation.url");
+        }
+
+        return undefined;
```

Comment:
Seems this logic is same as above, try to extract them to one function

Reply:
Updated, thanks!